nixos/modules/services/networking/wgautomesh.nix

```
@@ -0,0 +1,140 @@
+      default = "info";
+      description = mdDoc "wgautomesh log level (trace/debug/info/warn/error)";
+    };
+    interface = mkOption {
+      type = types.str;
+      description = mdDoc "Wireguard interface to manage (it is NOT created by wgautomesh, you should use another NixOS directive to create it such as `networking.wireguard.interfaces.wg0 = {...};`)";
+      example = "wg0";
+    };
+    gossipPort = mkOption {
+      type = types.port;
+      description = mdDoc "wgautomesh gossip port, this MUST be the same number on all nodes in the wgautomesh network";
+      default = 1666;
+    };
+    enableGossipEncryption = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc "Enable encryption of gossip traffic. If this is set, a common secret key must be copied at /var/lib/wgautomesh/gossip_secret on all nodes.";
+    };
+    enablePersistence = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc "Enable persistence of Wireguard peer info between restarts";
+    };
+    lanDiscovery = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc "Enable discovery of peers on the same LAN using UDP broadcast";
+    };
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc "Automatically open gossip port in firewall (recommended)";
+    };
+    upnpForwardPublicPort = mkOption {
+      type = types.nullOr types.port;
+      default = null;
+      description = mdDoc "Public port number to try to redirect to this machine's Wireguard daemon using UPnP IGD";
+    };
+    peers = mkOption {
+      type = types.listOf (types.submodule {
+        options = {
+          publicKey = mkOption {
+            type = types.str;
+            description = mdDoc "Wireguard public key of this peer";
+          };
+          address = mkOption {
+            type = types.str;
+            description = mdDoc "Wireguard address of this peer (a single IP address, multliple addresses or address ranges are not supported)";
+            example = "10.0.0.42";
+          };
+          endpoint = mkOption {
+            type = types.nullOr types.str;
+            description = mdDoc "Bootstrap endpoint for connecting to this Wireguard peer if no other address is known or none are working";
+            example = "wgnode.mydomain.example:51820";
+          };
+        };
+      });
+      description = mdDoc "wgautomesh peer list";
+    };
+  };
+
+  config = mkIf cfg.enable (let
+    peerDefs = map (peer:
+      let
+        endpointDef = if peer.endpoint == null then
+          ""
+        else
+          ''endpoint = "${peer.endpoint}"'';
+      in ''
+        [[peers]]
+        pubkey = "${peer.publicKey}"
+        address = "${peer.address}"
+        ${endpointDef}
+      '') cfg.peers;
+    extraDefs = (if cfg.lanDiscovery then [ "lan_discovery = true" ] else [ ])
```

```suggestion
    extraDefs = (optional cfg.lanDiscovery "lan_discovery = true")
```

All these `if` can be simplified by `optional`.